pkg/util/instanceselector/instanceselector.go

```diff
@@ -232,6 +233,21 @@ func findCheapestInstance(matches []InstanceData) string {
 	return cheapestInstance
 }
 
+func globToRegexp(globstr string) (*regexp.Regexp, error) {
+	if globstr == "" {
+		return nil, nil
+	}
+	// Need to escape regex special characters that aren't '*' and
+	// replace '*' with ".*"
+	globparts := strings.Split(globstr, "*")
+	for i := range globparts {
+		globparts[i] = regexp.QuoteMeta(globparts[i])
+	}
+	quotedString := strings.Join(globparts, ".*")
+	regexpstr := "^" + quotedString + "$"
+	return regexp.Compile(regexpstr)
+}
```

Comment:
There's a super simple Go package that implements glob matching: https://github.com/ryanuber/go-glob instead of trying to convert to a regexp and then do a (probably more expensive) regexp match.

Reply:
That's better.  I looked at another library that was not super-simple behind-the-scenes and rolled my own.  Using go-glob makes things simpler.